### PR TITLE
Remove Public Read header from presigned PUT

### LINKED ISSUE #104: can not upload via travis, Client Error 400
```
      _____          _
     / ____|        | |
    | |     ___   __| | ___  ___ _____   __
    | |    / _ \ / _  |/ _ \/ __/ _ \ \ / /
    | |___| (_) | (_| |  __/ (_| (_) \ V /
     \_____\___/ \____|\___|\___\___/ \_/
                                    v2.0.22
==> Detecting CI provider
    Travis Detected
==> Preparing upload
==> Processing gcov (disable by -X gcov)
    Executing gcov (['find', '/home/travis/build/bitranox/lib_programname', "-not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**'", '-type', 'f', '-name', '*.gcno', '', '-exec', 'gcov', '-pb', '', '{}', '+'])
find: unknown predicate `-not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**''
    Error running `['find', '/home/travis/build/bitranox/lib_programname', "-not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**'", '-type', 'f', '-name', '*.gcno', '', '-exec', 'gcov', '-pb', '', '{}', '+']`: Command '['find', '/home/travis/build/bitranox/lib_programname', "-not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**'", '-type', 'f', '-name', '*.gcno', '', '-exec', 'gcov', '-pb', '', '{}', '+']' returned non-zero exit status 1.
==> Collecting reports
    Generating coverage xml reports for Python
    + /home/travis/build/bitranox/lib_programname/coverage.xml bytes=5093
==> Appending environment variables
    + TRAVIS_OS_NAME
    + TRAVIS_PYTHON_VERSION
==> Uploading
    .url https://codecov.io
    .query commit=aef7044133d69d1e31476ff5f7bdfca078b0ee8d&branch=master&job=688009670&pr=false&service=travis&build=29.3&slug=bitranox%2Flib_programname&package=py2.0.22
    Pinging Codecov...
    Uploading to S3...
Error: 400 Client Error: Bad Request for url: https://storage.googleapis.com/codecov/v4/raw/2020-05-17/6FCEA983E832F40CEEBF26556F35B890/aef7044133d69d1e31476ff5f7bdfca078b0ee8d/2858a4a4-1a2d-4e23-98e0-229b2dfd6f96.txt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=GOOG1EQX6OZVJGHKK3633AAFGLBUCOOATRACRQRQF6HMSMLYUP6EAD6XSWAAY%2F20200517%2FUS%2Fs3%2Faws4_request&X-Amz-Date=20200517T104125Z&X-Amz-Expires=10&X-Amz-SignedHeaders=host&X-Amz-Signature=9a43bf2beca931a6e204941a97be43034d7339e65fdaf2f4030e29493df0af90

```

### PR DESCRIPTION
**Pull request by hootener**

### CODE REVIEW

**Comment by hootener** [comment]
This seems to be failing in CI with:

```
Restoring NuGet package Cake.0.32.1.
  GET https://api.nuget.org/v3-flatcontainer/cake/0.32.1/cake.0.32.1.nupkg
  OK https://api.nuget.org/v3-flatcontainer/cake/0.32.1/cake.0.32.1.nupkg 12ms
Installing Cake 0.32.1.
Adding package 'Cake.0.32.1' to folder '/home/appveyor/projects/codecov-exe/tools'
Added package 'Cake.0.32.1' to folder '/home/appveyor/projects/codecov-exe/tools'
Error: The assembly name is invalid.
Command exited with code 1
Build failed
```
And I just don't possess the knowledge of this application to make a fix here. Please let me know if I can help, @AdmiringWorm and @larzw

File: Source/Codecov/Upload/CodecovUploader.cs [hunk 1]
```diff
@@ -42,13 +42,9 @@ protected virtual void ConfigureContent(HttpContent content)
             content.Headers.ContentType = new MediaTypeHeaderValue("application/x-gzip");
         }
 
-        protected virtual void ConfigureRequest(HttpRequestMessage request)
```
Comment (AdmiringWorm) [inline Source/Codecov/Upload/CodecovUploader.cs:45]: Unfortunately, this method can not be removed entirely, as it is overridden in the fallback uploader to set the needed headers for the request there.

You could change the method to be an empty statement instead:
```cs
protected virtual void ConfigureRequest(HttpRequestMessage request) { }
```

**Comment by AdmiringWorm** [comment]
@hootener thank you for opening this PR. I was not aware that the storage location for S3 had been changed to something else as thus not needing this request header anymore.

Regarding the CI error, this is something I have been struggling to figure out myself as well, but so far have been unsuccessful in finding out why the failures happen.
The build only fails on Unix, and I believe there may be an incompatibility with the mono version on the CI server compared to earlier versions.

I may need to rewrite the build scripts to before I can merge in this PR.

File: Source/Codecov/Upload/CodecovUploader.cs (hunk 1 above)
Comment (hootener) [reply Source/Codecov/Upload/CodecovUploader.cs:45]: Modified this PR to take your recommendation, here. Please let me know if that's sufficient.

**Comment by hootener** [comment]
> I may need to rewrite the build scripts to before I can merge in this PR.

@AdmiringWorm Got it. Please let me know if there's anything else I need to do with respect to this PR or to unblock this process. Thanks!

**Comment by sn1020** [comment]
@hootener Might be helpful to add a readme update to this PR, similar to https://github.com/codecov/codecov-node/pull/181?

**Comment by hootener** [comment]
Done @sn1020 .

@AdmiringWorm I see this CI on this is passing now. Is it good to merge and release version 1.11.0?

**Comment by AdmiringWorm** [comment]
@hootener Unfortunately not yet, still working on a fixed build script.
The check succeeded because only the GitHub action ran (which unfortunately do not do any integration test yet), not the appveyor one which is the main one (it doesn't run on pull requests unfortunately due to missing the pull request event hook).

So I need to be sure that it works before merging this one in.

**Review by AdmiringWorm — APPROVED** [review]
LGTM

**Comment by AdmiringWorm** [comment]
@hootener your changes have been merged, thanks for your contribution 👍

**Comment by thomasrockhu** [comment]
Thanks @AdmiringWorm!

**Pull request merged**